src/get_versions.py

```diff
@@ -36,27 +36,52 @@ def get_versions(default=DEFAULT, verbose=False):
         ver = versions_from_keywords_f(vcs_keywords, tag_prefix)
         if ver:
             if verbose: print("got version from expanded keyword %s" % ver)
-            return ver
+            if not pep440_compliant:
+                return ver
+            else:
+                return rep_by_pep440(ver)
 
     ver = versions_from_file(versionfile_abs)
     if ver:
         if verbose: print("got version from file %s %s" % (versionfile_abs,ver))
-        return ver
+        if not pep440_compliant:
+            return ver
+        else:
+            return rep_by_pep440(ver)
 
     versions_from_vcs_f = vcs_function(VCS, "versions_from_vcs")
     if versions_from_vcs_f:
         ver = versions_from_vcs_f(tag_prefix, root, verbose)
         if ver:
             if verbose: print("got version from VCS %s" % ver)
-            return ver
+            if not pep440_compliant:
+                return ver
+            else:
+                return rep_by_pep440(ver)
 
     ver = versions_from_parentdir(parentdir_prefix, root, verbose)
     if ver:
         if verbose: print("got version from parentdir %s" % ver)
-        return ver
+        if not pep440_compliant:
+            return ver
+        else:
+            return rep_by_pep440(ver)
 
     if verbose: print("got version from default %s" % default)
     return default
 
 def get_version(verbose=False):
     return get_versions(verbose=verbose)["version"]
+
+
+def git2pep440(ver_str):
+    try:
+        tag, commits, _ = ver_str.split('-', 2)
```

Comment:
+1 on @tacaswell's proposal. I don't think there's any need to make it optional, not more than in the current versioneer behaviour.